### PR TITLE
Rework configuration

### PR DESCRIPTION
**Pull request by icanhazstring**
This will rework configuration by replacing yml by simple php config
arrays. Also the DI got a minor improvement and is now used accross the
project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 vendor/
+config/**/*.config.php
```
Comment (CiderAndWhisky) [inline .gitignore:2]: ??

Comment (icanhazstring) [reply .gitignore:2]: Look at the files :)
There is no yml file anymore, there are `config/.../jira.config.php.dist` files which will be replaced by tempa to something like `jira.config.php` which i don't want to be checked into version control

Comment (CiderAndWhisky) [reply .gitignore:2]: Ah, jetzt dist files. Verstehe.

**Review by CiderAndWhisky — APPROVED** [review]

**Review by demichl68 — APPROVED** [review]
This will defo. help maintaining the repo 👍

**Pull request merged**